### ISSUE TITLE
Role ARN config option isn't working

### ISSUE THREAD

**Issue by autarchprinceps** [logstash-plugins/logstash-input-kinesis#65]
I added a role_arn as per the documented configuration option, but it is still trying to use the default credentials.
I checked and my version of the plugin 2.0.11 should include that ability.

kinesis {
    id => "input.kinesis.1"
    region => "_region_"
    kinesis_stream_name => "_stream_"
    metrics => "cloudwatch"
    application_name => "_table_"
    role_arn => "arn:aws:iam::_account_:role/_rolename_"
    tags => ["input_kinesis", "codec_cloudwatch_logs"]
    codec => cloudwatch_logs
  }

But I get a com.amazonaws.services.kinesis.leases.exceptions.DependencyException: com.amazonaws.services.dynamodbv2.model.AmazonDynamoDBException: User: _default credentials_ is not authorized to perform: dynamodb:DescribeTable on resource: arn:aws:dynamodb:eu-central-1:_account_:table/_custom table_ (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: AccessDeniedException)

Since I do get the non-standard table name that I overrode in the input config, I am confident, that the configuration applies, but it does switch the role. There are no prior failures or interesting messages in the logs and afterwards it repeats the same access denied error for other dynamodb api calls.

**Comment by robbavey** [comment]
@autarchprinceps Do you have sufficient permissions for a Kinesis consumer - see https://docs.aws.amazon.com/streams/latest/dev/learning-kinesis-module-one-iam.html

**Comment by autarchprinceps** [comment]
As I wrote, it is not using the role I configured with the rights. It explicitly states that the instance profile role ARN is the one used. "User: _insert instance profile credentials here_ is not authorized" - That's why I gave it an alternative role via the plugin configuration. Yes, that role has sufficient rights, but even if it didn't the message would be "User: _insert override credentials from plugin settings here_ is not authorized". The fact that it tries to use the wrong role is the reason it is not authorized, which the message explicitly states, therefore it is not assuming that role before or at least not using the assumed role for making the dynamodb calls.
